internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -121,25 +124,33 @@ func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, s c
 	}
 
 	for name, status := range in.DependsOn {
-		if !isValidStatus(status) {
-			return errInvalidDependsOnStatus
+		if ok, err := isValidStatus(status, name, s); !ok {
+			return err
 		}
-		if isEssential(name, s) && !isEssentialStatus(status) {
-			return errEssentialContainerStatus
+		if ok, err := isEssentialStatus(status, name, s); isEssential(name, s) && !ok {
```

Comment:
Can we add a quick comment here explaining this bit of logic? 
```go
// Container cannot depend on essential containers having status `complete` or `success`
```

Reply:
Added a clarifying comment.